util/types/datum.go

```
@@ -1609,10 +1571,9 @@ func CoerceDatum(sc *variable.StatementContext, a, b Datum) (x, y Datum, err err
 			x.SetFloat64(float64(x.GetInt64()))
 		case KindUint64:
 			x.SetFloat64(float64(x.GetUint64()))
-		case KindMysqlHex:
-			x.SetFloat64(x.GetMysqlHex().ToNumber())
-		case KindMysqlBit:
-			x.SetFloat64(x.GetMysqlBit().ToNumber())
+		case KindBinaryLiteral, KindMysqlBit:
+			val, _ := x.GetBinaryLiteral().ToInt()
```

error is not handled.